scene/main/node.cpp

```diff
@@ -1814,14 +1814,28 @@ Node *Node::get_child(int p_index, bool p_include_internal) const {
 
 TypedArray<Node> Node::get_children(bool p_include_internal) const {
 	ERR_THREAD_GUARD_V(TypedArray<Node>());
-	TypedArray<Node> arr;
-	int cc = get_child_count(p_include_internal);
-	arr.resize(cc);
-	for (int i = 0; i < cc; i++) {
-		arr[i] = get_child(i, p_include_internal);
+	_update_children_cache();
+
+	TypedArray<Node> children;
+
+	uint64_t start;
+	uint64_t end;
+	if (p_include_internal) {
+		start = 0;
+		end = data.children_cache.size();
+	} else {
+		start = data.internal_children_front_count_cache;
+		end = data.children_cache.size() - data.internal_children_back_count_cache + start;
```

Comment:
Looking at this again. I think this needs to be:

```suggestion
		end = data.children_cache.size() - data.internal_children_back_count_cache;
```